### PR TITLE
tasks: New task on getting started with RN mobile app.

### PR DESCRIPTION
**Pull request by jainkuniya**
Fix:#615

### CODE REVIEW

File: tasks/2017/intro-to-zulip-mobile.md [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+Run app using
+```
+react-native run-ios // on Mac OS, to run on iOS device (simulator or device)
+react-native run-android // to run on Android device (emulator or device)
+```
+
+If you are running on Android emulator, then make sure it is launched before you run command on terminal.
+
+Enter server url (we recommend to use https://chat.zulip.org) and your login credentials.
+Open left drawer and narrow to `test here` stream and send message with your name as topic. And share your experience with us in `mobile` stream with topic as `feedback: setting up environment | <your name>` (replace `<your name>` with your name).
```
Comment (lonerz) [inline tasks/2017/intro-to-zulip-mobile.md]: and send **a** message with your name as **the** topic

File: tasks/2017/intro-to-zulip-mobile.md [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+
+Now task is to share your learning and updating [docs](https://github.com/zulip/zulip-mobile/tree/master/docs)
```
Comment (lonerz) [inline tasks/2017/intro-to-zulip-mobile.md]: Not sure what the task is supposed to be?

File: tasks/2017/intro-to-zulip-mobile.md [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+
+**Tips:**
+After running app, answering following questions will help you in getting ideas what to update in doc.
+* Does existing docs are sufficient to set up developing environment? If no, then what is lacking else suggest changes so that we can improve them.
+* Does you face any problem while setting up? Any error in terminal or anything other. Add such errors with solutions in docs.
+
+### Submit
+
+Submit the three screenshots of running app on simulator using the GCI tasks interface.
```
Comment (lonerz) [inline tasks/2017/intro-to-zulip-mobile.md]: Three screenshots of? You should probably specify what things students should be taking screenshots of.

**Comment by jainkuniya** [comment]
Thanks @lonerz for quick review :)
Updated 👍 
I am newbie here in creating tasks :), let me know other changes :)

File: tasks/2017/intro-to-zulip-mobile.md [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+
+* Update your working copy of [zulip-mobile](https://github.com/zulip/zulip-mobile) and then create a feature branch. [Learn
+  how](../../before-every-task.md).
+
+### Practice Running zulip-mobile
+
+Run app using
+```
+react-native run-ios // on Mac OS, to run on iOS device (simulator or device)
+react-native run-android // to run on Android device (emulator or device)
```
Comment (synicalsyntax) [inline tasks/2017/intro-to-zulip-mobile.md]: `react-native run-android // emulate or run the app on an Android device`

File: tasks/2017/intro-to-zulip-mobile.md [hunk 5]
```diff
@@ -0,0 +1,59 @@
...
+run command on terminal.
+
+Enter server url (we recommend to use https://chat.zulip.org) and your login credentials.
+Open left drawer and narrow to `test here` stream and send a message with your
+name as the topic. And share your experience with us in `mobile` stream with
+topic as `feedback: setting up environment | <your name>` (replace `<your name>`
+with your name).
+
+Now task is to
+* share your learning with us on https://chat.zulip.org/#narrow/stream/mobile
```
Comment (synicalsyntax) [inline tasks/2017/intro-to-zulip-mobile.md]:
```md
share what you have learned with the Zulip mobile team on the [#mobile](https://chat.zulip.org/#narrow/stream/mobile) stream
````

File: tasks/2017/intro-to-zulip-mobile.md [hunk 6]
```diff
@@ -0,0 +1,59 @@
...
+If you are running on Android emulator, then make sure it is launched before you
+run command on terminal.
+
+Enter server url (we recommend to use https://chat.zulip.org) and your login credentials.
+Open left drawer and narrow to `test here` stream and send a message with your
+name as the topic. And share your experience with us in `mobile` stream with
+topic as `feedback: setting up environment | <your name>` (replace `<your name>`
+with your name).
+
+Now task is to
```
Comment (synicalsyntax) [inline tasks/2017/intro-to-zulip-mobile.md]: Now, your task is to:

File: tasks/2017/intro-to-zulip-mobile.md [hunk 7]
```diff
@@ -0,0 +1,59 @@
...
+* share your learning with us on https://chat.zulip.org/#narrow/stream/mobile
+* updating [docs](https://github.com/zulip/zulip-mobile/tree/master/docs)
```
Comment (synicalsyntax) [inline tasks/2017/intro-to-zulip-mobile.md]: `update the [documentation](https://github.com/zulip/zulip-mobile/tree/master/docs) with several improvements`

File: tasks/2017/intro-to-zulip-mobile.md [hunk 8]
```diff
@@ -0,0 +1,59 @@
...
+* take screenshot of the app.
```
Comment (synicalsyntax) [inline tasks/2017/intro-to-zulip-mobile.md]: How many screenshots? Maybe multiple to show that they can navigate the interface fairly easily?

File: tasks/2017/intro-to-zulip-mobile.md [hunk 9]
```diff
@@ -0,0 +1,59 @@
...
+
+**Tips:**
+After running app, answering following questions will help you in getting ideas
+what to update in doc.
+* Does existing docs are sufficient to set up developing environment? If no,
+then what is lacking else suggest changes so that we can improve them.
+* Does you face any problem while setting up? Any error in terminal or anything
```
Comment (synicalsyntax) [inline tasks/2017/intro-to-zulip-mobile.md]: Did you face any problems while setting up your development environments (e.g. errors in the command line)? Note these errors along with their solutions in your docs.

File: tasks/2017/intro-to-zulip-mobile.md [hunk 10]
```diff
@@ -0,0 +1,59 @@
...
+other. Add such errors with solutions in docs.
+
+### Submit
+
+After running app,
+
+* narrow to `all private messages` from selecting envelop icon in drawer.
+Take screenshot of the simulator/emulator window.
+
+* open drawer and switch to night mode.
```
Comment (synicalsyntax) [inline tasks/2017/intro-to-zulip-mobile.md]: see my other comments about the drawer.

File: tasks/2017/intro-to-zulip-mobile.md [hunk 11]
```diff
@@ -0,0 +1,59 @@
...
+Take screenshot of the simulator/emulator window.
```
Comment (synicalsyntax) [inline tasks/2017/intro-to-zulip-mobile.md]: Take a screenshot

File: tasks/2017/intro-to-zulip-mobile.md [hunk 12]
```diff
@@ -0,0 +1,59 @@
...
+then what is lacking else suggest changes so that we can improve them.
+* Does you face any problem while setting up? Any error in terminal or anything
+other. Add such errors with solutions in docs.
+
+### Submit
+
+After running app,
+
+* narrow to `all private messages` from selecting envelop icon in drawer.
+Take screenshot of the simulator/emulator window.
```
Comment (synicalsyntax) [inline tasks/2017/intro-to-zulip-mobile.md]: Take a screenshot

File: tasks/2017/intro-to-zulip-mobile.md [hunk 13]
```diff
@@ -0,0 +1,59 @@
...
+
+* open drawer and switch to night mode.
+Take screenshot of the simulator/emulator window.
+
+* narrow to `test here` stream, and start composing new message. Type `:smi` or
```
Comment (synicalsyntax) [inline tasks/2017/intro-to-zulip-mobile.md]: narrow to the `test here` stream, and start composing a new message.

File: tasks/2017/intro-to-zulip-mobile.md [hunk 14]
```diff
@@ -0,0 +1,59 @@
...
+any other emoji of your choice.
+Take screenshot of the simulator/emulator window. Make sure emoji autocomplete
```
Comment (synicalsyntax) [inline tasks/2017/intro-to-zulip-mobile.md]: Take a screenshot

Make sure the emoji autocomplete* is visible in the screenshot

*is this feature a popover or a modal or something? It should be mentioned here.

File: tasks/2017/intro-to-zulip-mobile.md [hunk 15]
```diff
@@ -0,0 +1,59 @@
...
+is visible in the screenshot.
+
+
+(make sure simulator/emulator window is included in the screenshots)
```
Comment (synicalsyntax) [inline tasks/2017/intro-to-zulip-mobile.md]: I think you've already stated that you need to take a picture of the simulator/emulator window before :)

File: tasks/2017/intro-to-zulip-mobile.md [hunk 16]
```diff
@@ -0,0 +1,59 @@
...
+topic as `feedback: setting up environment | <your name>` (replace `<your name>`
+with your name).
+
+Now task is to
+* share your learning with us on https://chat.zulip.org/#narrow/stream/mobile
+* updating [docs](https://github.com/zulip/zulip-mobile/tree/master/docs)
+* take screenshot of the app.
+
+**Tips:**
+After running app, answering following questions will help you in getting ideas
```
Comment (synicalsyntax) [inline tasks/2017/intro-to-zulip-mobile.md]: After running the app, answering the following questions will help you write your docs:

**Comment by jainkuniya** [comment]
Thanks @synicalsyntax .
Updated 👍

File: tasks/2017/intro-to-zulip-mobile.md (hunk 8 above)
Comment (jainkuniya) [reply tasks/2017/intro-to-zulip-mobile.md]: added more info.

File: tasks/2017/intro-to-zulip-mobile.md [hunk 17]
```diff
@@ -0,0 +1,59 @@
...
+* Does existing docs are sufficient to set up developing environment? If no,
+then what is lacking else suggest changes so that we can improve them.
+* Does you face any problem while setting up? Any error in terminal or anything
+other. Add such errors with solutions in docs.
+
+### Submit
+
+After running app,
+
+* narrow to `all private messages` from selecting envelop icon in drawer.
```
[the remark this answers is not in the document]
Comment (jainkuniya) [reply tasks/2017/intro-to-zulip-mobile.md]: I thinks drawer is well known.
BTW added instructions on how to open drawer.

**Comment by cdikibo** [comment]
@jainkuniya react-native run-ios has never worked for me since participating on the iOS app. npm run ios-min and npm run ios-max work 100%

**Comment by timabbott** [comment]
Thanks for all your work on this @jainkuniya and reviewers!  I did a pass of editing (in ebc3e74401da0ced95696d58475735f90eab3269) and merged this.  I think there's a few follow-up things:

* Figure out whether we need to change `react-native run` to `npm run` as @cdikibo suggests.
* I think we should add a bunch of links to intro tutorials for React, RN, the Android simulator, etc. in the prereq section of this task (or maybe just links to where we link to those in the zulip-mobile docs).  It feels like we could make this a lot easier for students who are excited and have time to learn, but don't have RN or mobile experience, to find good resources for learning how things work.
* We need to create a scripts/ script for creating ~50 copies of this task.  I wouldn't tag the script as "beginner", since I think we want only more experienced students to do this one (I added a little bit of a mild scare note to that effect in the description).
* Just noticed, I failed to fill in the links to the git-comfortable and submit-a-pull-request tasks...

@jainkuniya can you make sure these things happen by EOD tomorrow so that this task is ready to go when GCI starts?

**Comment by timabbott** [comment]
@saketkumar95 can you get a friend to attempt to follow the instructions in this task in the next day?

**Comment by saketkumar95** [comment]
@timabbott  Yaa sure! I'll tell @phunsukwangdu to attempt this.

**Comment by jainkuniya** [comment]
Submitted https://github.com/zulip/zulip-gci/pull/651

**Comment by saketkumar95** [comment]
@jainkuniya Can you also assign me this task. I would be happy to help. 😄

**Comment by vrinda6247** [comment]
@timabbott @saketkumar95 I followed the instructions in this task. It was really great. I didn't faced any such problems as I also work on react native and It didn't created any such problems. I doubt if someone who hasn't worked on react native earlier may get any npm or react native errors first time?

**Pull request closed without merging**